app/script/util/TimeUtil.js

```diff
@@ -80,41 +96,51 @@ z.util.TimeUtil = {
 
   /**
    * Format milliseconds into 15s, 2m.
-   * @note Implementation based on: https://gist.github.com/deanrobertcook/7168b38150c303a2b4196216913d34c1
    * @param {number} duration - Duration to format in milliseconds
-   * @param {boolean} rounded - Enables rounding of numbers
-   * @param {number} maximumUnits - Maximum number of units shown in the textual representation
-   * @returns {Object} Unit, value and localized string
+   * @returns {DurationUnit} Unit, value and localized string
    */
-  formatDuration: (duration, rounded = true, maximumUnits = 1) => {
-    const mappedUnits = z.util.TimeUtil.durationUnits().map((unit, index) => {
-      let value = duration;
-      if (index > 0) {
-        value %= z.util.TimeUtil.durationUnits()[index - 1].value;
-      }
-      value /= unit.value;
-      value = rounded && value >= 1 ? Math.round(value) : Math.floor(value);
-      const longUnit = z.string[value === 1 ? unit.singular : unit.plural];
-      return {
-        longUnit,
-        unit: unit.unit,
-        value,
-      };
-    });
-
+  formatDuration: duration => {
+    const mappedUnits = z.util.TimeUtil.mapUnits(duration, true);
     const firstNonZeroUnit = mappedUnits.find(unit => unit.value > 0);
-    const startIndex = mappedUnits.indexOf(firstNonZeroUnit);
-    const validUnits = mappedUnits.slice(startIndex, startIndex + maximumUnits);
-    const longText = validUnits.map(unit => `${unit.value} ${z.l10n.text(unit.longUnit)}`).join(', ');
-    const upperUnit = validUnits[0] || {};
-
     return {
-      text: longText,
-      unit: upperUnit.unit,
-      value: upperUnit.value,
+      text: `${firstNonZeroUnit.value} ${z.l10n.text(firstNonZeroUnit.longUnit)}`,
+      unit: firstNonZeroUnit.unit,
+      value: firstNonZeroUnit.value,
     };
   },
 
+  /**
+   * Generate a human readable string of the remaining time
+   * @param {number} duration - the remaining time in milliseconds
+   * @returns {string} readable representation of the remaining time
+   */
+  formatDurationCaption: duration => {
+    const mappedUnits = z.util.TimeUtil.mapUnits(duration, false);
+    const hours = mappedUnits.find(unit => unit.unit === 'h');
+    const minutes = mappedUnits.find(unit => unit.unit === 'm');
+    const hasHours = hours.value > 0;
+    const validUnitStrings = [];
+    for (let index = 0; index < mappedUnits.length; index++) {
```

Comment:
Any particular reason we are not using a `for...of` here?

Reply:
A couple lines down I need the index to get the following element.